subprojects/logging/src/main/java/org/gradle/api/logging/configuration/LoggingConfiguration.java

```diff
@@ -45,6 +45,17 @@
     void setConsoleOutput(ConsoleOutput consoleOutput);
 
     /**
+     * Returns whether dry-run mode is enabled or not.
+     * Defaults to false.
+     */
+    boolean isDryRun();
+
+    /**
+     * Enables or disables dry-run (log but do not perform task actions) mode.
+     */
+    void setDryRun(boolean dryRun);
```

Comment:
I think too and the execution goes through in `WithLogging` class. At this point. we aren't parsing the entire `StartParameter`. I don't know what would be the best choice for this. We could do like the parallel configuration and extra it to its own configuration class if you like.